runtime/src/rpc.rs

```diff
@@ -1301,15 +1307,72 @@ impl BtcRelayPallet for PolkaBtcProvider {
         Ok(self.ext_client.stable_bitcoin_confirmations(head).await?)
     }
 
+    /// Get the global security parameter for stable parachain confirmations
+    async fn get_parachain_confirmations(&self) -> Result<BlockNumber, Error> {
+        let head = self.get_latest_block_hash().await?;
+        Ok(self.ext_client.stable_parachain_confirmations(head).await?)
+    }
+
     /// Wait until Bitcoin block is submitted to the relay
-    async fn wait_for_block_in_relay(&self, block_hash: H256Le, num_confirmations: u32) -> Result<(), Error> {
+    async fn wait_for_block_in_relay(
+        &self,
+        block_hash: H256Le,
+        btc_confirmations: Option<BlockNumber>,
+    ) -> Result<(), Error> {
+        let get_bitcoin_confirmations = async {
+            if let Some(btc_confirmations) = btc_confirmations {
+                Ok(btc_confirmations)
+            } else {
+                self.get_bitcoin_confirmations().await
+            }
+        };
+
+        let (bitcoin_confirmations, parachain_confirmations) =
+            futures::future::try_join(get_bitcoin_confirmations, self.get_parachain_confirmations()).await?;
+
+        async fn has_sufficient_confirmations(
```

Comment:
probably could be a closure, it looks weird to me to have a function inside a function. But you can also leave as-is if you prefer

Reply:
yeah I'll maybe refactor this later, should give us what we want for now